pkgs/by-name/co/compiler-explorer/package.nix

```diff
@@ -0,0 +1,144 @@
+      clang_11 clang_12 clang_13 clang_14 clang_15 clang_16 clang_17 lld_17 llvmPackages_17;
+    isC = true;
+  };
+
+  rustConfigFile = import ./cfg/rust.nix {
+    inherit defaultGcc rustc rustfilt writeText utils;
+  };
+
+  goConfigFile = import ./cfg/go.nix {
+    inherit defaultGcc gccgo13 writeText utils;
+  };
+
+  pythonConfigFile = import ./cfg/python3.nix {
+    inherit defaultGcc python3Full writeText utils;
+  };
+
+  ceConfigFile = import ./cfg/compiler-explorer.nix {
+    inherit lib writeText utils python3Full defaultGcc cmake coreutils;
+  };
+  executionConfigFile = import ./cfg/compiler-explorer-execution.nix {
+    inherit writeText utils;
+  };
+
+in
+buildNpmPackage rec {
+
+  pname = "compiler-explorer";
+  version = "10723";
+  src = fetchFromGitHub {
+    owner = "compiler-explorer";
+    repo = "compiler-explorer";
+    rev = "gh-${version}";
+    hash = "sha256-e8G1WNII61FWGscxSTlB2GvDP5e3945AslDP2h4jr2k=";
+  };
+  npmDepsHash = "sha256-8e8JtVeJXx1NxwYlN4SRJB2K/RJv9plnAwlRNWHWD1M=";
+
+  nativeBuildInputs = [ nodejs_20 typescript husky rsync ];
+  dontNpmBuild = true;
+  dontNpmInstall = true;
+  CYPRESS_INSTALL_BINARY = 0;
+  NODE_DIR = "${nodejs_20}";
+  postPatch = ''
+    rm Makefile
+    cp etc/config/sponsors.yaml .
+    cp etc/config/site-templates.conf .
+    rm -rf etc/config/*
+    mv sponsors.yaml etc/config/
+    mv site-templates.conf etc/config/
+  '';
+
+  buildPhase = ''
+    rm -rf out
+    rm -rf lib/storage/data
+    mkdir -p out/dist
+    cp -R etc out/dist/
+    cp -R examples out/dist/
+    cp -R views out/dist/
+    cp -R types out/dist/
+    cp -R package.json out/dist/
+    cp -R package-lock.json  out/dist/
+
+    # Set up and build and webpack everything
+    npm install --no-audit
+    npm run webpack
+    npm run ts-compile
+
+    # Now install only the production dependencies in our output directory
+    cd out/dist
+    npm install --no-audit --ignore-scripts --production
+    #rm -rf node_modules/.cache/ node_modules/monaco-editor/
+    #find node_modules -name \*.ts -delete
+
+    # Output some magic for GH to set the branch name and release name
+
+    # Run to make sure we haven't just made something that won't work
+    node --no-warnings=ExperimentalWarning --loader ts-node/esm ./app.js --version --dist
+
+    cd ../..
+    rsync -av out/dist/ $out/
+    rsync -av out/webpack/static/ $out/static/
+
+  '';
+
+
+   installPhase = ''
+     cd $out
+     cp ${ceConfigFile} etc/config/compiler-explorer.defaults.properties
+     cp ${executionConfigFile} etc/config/execution.defaults.properties
+     cp ${cConfigFile} etc/config/c.defaults.properties
+     cp ${cppConfigFile} etc/config/c++.defaults.properties
+     cp ${rustConfigFile} etc/config/rust.defaults.properties
+     cp ${pythonConfigFile} etc/config/python.defaults.properties
+     cp ${goConfigFile} etc/config/go.defaults.properties
```

Comment:
I think configuration of the compilers is better suited to a Nixos module. This will let users configure these without recompiling the package.

Reply:
Wouldn't that prevent people from using compiler-explorer on non-nixos? (Since the package would be an empty shell)

Maybe we should expect some inputs to be overwritten?

Reply:
Fair point. Maybe we can bundle a configuration for non Nixos which uses compilers on PATH or from /use/bin. Their defaults already do this. Nixos can then override this with the module system.

Reply:
> Fair point. Maybe we can bundle a configuration for non Nixos which uses compilers on PATH or from /use/bin. Their defaults already do this. Nixos can then override this with the module system.

One of the points of compiler-explorer is to check different compiler versions, and it seems like nix is very good at just that. I wouldn't love having to rely on tools outside nix, which kind of defeats the point.

Maybe the best approach would be to see if compiler-explorer allows to find the configuration files in a random folder via a parameter or a env var?

That could solve the issue for non-nixos users with home-package, so a home-manager module could use it (Or an user manually).

Reply:
[should](https://github.com/compiler-explorer/compiler-explorer/blob/1e72627cb7b9c61860f0588069cb0dd3ed2807d1/test/handlers/site-templates-tests.ts#L7)

It seems like it could be done without much hassle